proof/proof.go

```diff
@@ -7,10 +7,25 @@ package proof
 
 import (
 	"errors"
+	"hash"
 
 	"github.com/dedis/crypto"
 )
 
+// NOTE This suite is the same as the global one. Writing it is not an easy when you
+// don't know what methods you want to have (i.e.  the same set of methods that
+// edwards25519 implements etc).
+// dleq.go uses this interface while it does not need it all of it, so we're back to same
+// problem as before. Solution is to move it out to an external package, and
+// that brings an additional amount of work for "giving out to a package only
+// what it needs".
+type Suite interface {
```

Comment:
Or if you really want to go the hard way, take a piece of paper, go through every function, and write down what they need.